src/Juvix/Core/EAC/ConstraintGen.hs

```diff
@@ -196,15 +203,16 @@ boxAndTypeConstraint parameterizedAssignment term = do
 
 -- Generate constraints.
 generateTypeAndConstraints ∷
+  ∀ m primTy primVal.
   ( HasState "path" Path m,
     HasState "varPaths" VarPaths m,
     HasState "nextParam" Param m,
-    HasState "typeAssignment" TypeAssignment m,
+    HasState "typeAssignment" (TypeAssignment primTy) m,
```

Comment:
Some type aliases might be nice here?

Reply:
How so? They would need to be scoped to the function.